### PR TITLE
Rename configuration rename-rules to acronym-mapping

### PR DESCRIPTION
**Pull request by RodgeFu**
Fixes https://github.com/Azure/autorest.csharp/issues/2456

### CODE REVIEW

**Review by ArcturusZhang — COMMENTED** [review]
I suggest we do it in two PRs in the generator:
1. open a generator PR (**the first PR**) to make the generator to accept both `rename-mapping` and `acronym-mapping` as keys (`acronym-mapping` should have higher priority)
2. the regen of the first PR should be no change.
3. we could merge the first PR.
4. open a PR (**the SDK PR**) in the azure-sdk-for-net repo to rename all config to `acronym-mapping`. There should also be no code change when doing this. Merge this PR.
5. open another generator PR (**the second PR**) to make the generator only accept `acronym-mapping`. The regen of this PR should also contain no change.

I personally prefer we should do **the second PR** later - maybe months later, because there might be ongoing PR waiting to merge into the SDK if our generator no longer supports `rename-mapping`, they will not get a notice, and the generator might just work fine.

Also we need to change the mgmt template as well to use the new config name in **the SDK PR**: https://github.com/Azure/azure-sdk-for-net/blob/858b7ba9da51891ccb77b7aa88d94a44b9bffeb2/eng/templates/Azure.ResourceManager.Template/src/autorest.md?plain=1#L26

File: src/AutoRest.CSharp/Mgmt/AutoRest/MgmtConfiguration.cs [hunk 1]
```diff
@@ -265,7 +265,7 @@ internal static MgmtConfiguration GetConfiguration(IPluginCommunication autoRest
                 operationPositions: autoRest.GetValue<JsonElement?>("operation-positions").GetAwaiter().GetResult(),
                 requestPathToSingletonResource: autoRest.GetValue<JsonElement?>("request-path-to-singleton-resource").GetAwaiter().GetResult(),
                 overrideOperationName: autoRest.GetValue<JsonElement?>("override-operation-name").GetAwaiter().GetResult(),
-                renameRules: autoRest.GetValue<JsonElement?>("rename-rules").GetAwaiter().GetResult(),
+                renameRules: GetAcronymMappingConfig(autoRest),
```
Comment (ArcturusZhang) [inline src/AutoRest.CSharp/Mgmt/AutoRest/MgmtConfiguration.cs:268]: Could we rename the name of the argument here as well?
```suggestion
                acronymMapping: GetAcronymMappingConfig(autoRest),
```

File: docs/mgmt/polishing.md [hunk 2]
```diff
@@ -18,7 +18,7 @@
 A mechanism of renaming is introduced in the generator to unify the casing of some words.
```
Comment (ArcturusZhang) [inline docs/mgmt/polishing.md:18]: We should also change the name of this section

Comment (ArcturusZhang) [reply docs/mgmt/polishing.md:18]:
```suggestion
A mechanism of acronym mapping is introduced in the generator to unify the casing of acronyms.
```

**Review by ArcturusZhang — APPROVED** [review]
Approved with two comments

**Review by ArthurMa1978 — APPROVED** [review]

**Pull request merged**